src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkInterfacePal.Browser.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices.JavaScript;
+
+namespace System.Net.NetworkInformation
+{
+    internal static partial class BrowserNetworkInterfaceInterop
+    {
+        [JSImport("INTERNAL.network_wasm_online")]
+        public static partial bool IsOnline();
+
+        [JSImport("INTERNAL.network_wasm_add_change_listener")]
+        public static partial void AddChangeListener([JSMarshalAs<JSType.Function<JSType.Boolean>>] Action<bool> handler);
+    }
+
+    internal static class NetworkInterfacePal
+    {
+        public static NetworkInterface[] GetAllNetworkInterfaces() => throw new PlatformNotSupportedException();
+        public static int IPv6LoopbackInterfaceIndex => throw new PlatformNotSupportedException();
+        public static int LoopbackInterfaceIndex => throw new PlatformNotSupportedException();
```

Comment:
Why are these needed? Also, they don't seem to be used.

Reply:
They are used here https://github.com/dotnet/runtime/blob/2baf83fd2a162f75850885733bf63c2fcbd97c46/src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkInterface.cs#L19